common/tree/script_test.go

```diff
@@ -17,25 +18,294 @@ import (
+				fmtCode(txscript.OP_DROP) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey1 +
+				fmtCode(txscript.OP_CHECKSIGVERIFY) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey2 +
+				fmtCode(txscript.OP_CHECKSIG),
+			expectedError: true,
+		},
+
+		{
+			run: "CLTVMultisig missing OP_CHECKLOCKTIMEVERIFY",
+			script: exampleLocktime +
+				fmtCode(txscript.OP_DROP) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey1 +
+				fmtCode(txscript.OP_CHECKSIGVERIFY) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey2 +
+				fmtCode(txscript.OP_CHECKSIG),
+			expectedError: true,
+		},
+	}
+
+	var conditionMultisigClosureCases = []ClosureTestRun{
+		{
+			run: "Conditional multisig closure, ex condition 2 nums that add up to 8",
+			script: fmtCode(txscript.OP_5) +
+				fmtCode(txscript.OP_3) +
+				fmtCode(txscript.OP_2DUP) +
+				fmtCode(txscript.OP_ADD) +
+				fmtCode(txscript.OP_8) +
+				fmtCode(txscript.OP_EQUALVERIFY) +
+				fmtCode(txscript.OP_SUB) +
+				fmtCode(txscript.OP_2) +
+				fmtCode(txscript.OP_EQUAL) +
+				fmtCode(txscript.OP_VERIFY) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey1 +
+				fmtCode(txscript.OP_CHECKSIGVERIFY) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey2 +
+				fmtCode(txscript.OP_CHECKSIG),
+			expectedError: false,
+		},
+
+		{
+			run: "Conditional multisig closure, condition hash",
+			script: fmtCode(txscript.OP_1) +
+				fmtCode(txscript.OP_HASH160) +
+				fmtCode(txscript.OP_DATA_20) +
+				"628850CB844FE63C308C62AFC8BC5351F1952A7F" +
+				fmtCode(txscript.OP_EQUAL) +
+				fmtCode(txscript.OP_VERIFY) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey1 +
+				fmtCode(txscript.OP_CHECKSIGVERIFY) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey2 +
+				fmtCode(txscript.OP_CHECKSIG),
+			expectedError: false,
+		},
+
+		{
+			run: "Conditional multisig closure, missing OP_VERIFY",
+			script: fmtCode(txscript.OP_1) +
+				fmtCode(txscript.OP_HASH160) +
+				fmtCode(txscript.OP_DATA_20) +
+				"628850CB844FE63C308C62AFC8BC5351F1952A7F" +
+				fmtCode(txscript.OP_EQUAL) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey1 +
+				fmtCode(txscript.OP_CHECKSIGVERIFY) +
+				fmtCode(txscript.OP_DATA_32) +
+				exPubKey2 +
+				fmtCode(txscript.OP_CHECKSIG),
+			expectedError: true,
+		},
+	}
+
+	var cases = [][]ClosureTestRun{
+		multisigClosureCases,
+		csvMultisigClosureCases,
+		cltvMultisigClosureCases,
+		conditionMultisigClosureCases,
+	}
+
+	var mergedCases []ClosureTestRun
+	for _, c := range cases {
+		mergedCases = append(mergedCases, c...)
+	}
+
+	for _, testCase := range mergedCases {
+		t.Run(testCase.run, func(t *testing.T) {
+			scriptBytes, err := hex.DecodeString(testCase.script)
+			require.NoError(t, err)
+			closure, err := tree.DecodeClosure(scriptBytes)
+
+			if testCase.expectedError && closure != nil {
+				t.Errorf("Expected error but script decoded!")
+			}
+
+			if !testCase.expectedError && err != nil {
+				t.Errorf("Failed to decode correct script!")
+			}
```

Comment:
let's keep using `require` here:
```suggestion
			if testCase.expectedError {
			      require.Nil(t, closure)
			      require.Error(t, err)
			      continue
			}
			
			require.NoError(t, err)
			require.NotNil(t, closure)
```